### PR TITLE
NIFI-5847 Added GovCloud-East region to enum list.

### PR DESCRIPTION
**Pull request by jmlogan**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [x] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [x] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by jmlogan** [comment]
The Travis build failing seems unrelated -- is there a way to trigger a rebuild?

File: nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/src/main/java/org/apache/nifi/processors/aws/regions/AWSRegions.java [hunk 1]
```diff
@@ -18,7 +18,8 @@
 
 public enum AWSRegions {
 
-    GovCloud("us-gov-west-1", "AWS GovCloud (US)"),
+    GovCloud("us-gov-west-1", "AWS GovCloud West (US)"),
```
Comment (patricker) [inline nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/src/main/java/org/apache/nifi/processors/aws/regions/AWSRegions.java]: In the past, the display names have always been kept as what AWS shows. 
AWS docs still show:
 - AWS GovCloud (US-East)
 - AWS GovCloud (US)

Maybe it would be better to revert the change to `AWS GovCloud West (US)`? Then it will match the docs.

Comment (patricker) [reply nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/src/main/java/org/apache/nifi/processors/aws/regions/AWSRegions.java]: I'm going to correct myself. Some docs say (US). Some say, `(US-West)` and `(US-East)`.

https://docs.aws.amazon.com/govcloud-us/index.html#lang/en_us

So maybe that would be better?

Comment (jmlogan) [reply nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/src/main/java/org/apache/nifi/processors/aws/regions/AWSRegions.java]: I can change the display name -- what about the enum itself though? Changing the enum would break any existing flows I think? Just leave it so that GovCloud is implicitly West?

Comment (patricker) [reply nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/src/main/java/org/apache/nifi/processors/aws/regions/AWSRegions.java]: Just the display name is all I'm worried about, everything else looks fine.

**Comment by jmlogan** [comment]
I can close it but I was wondering how the SDK derives that list ...it
looks like it's just an enum on their side -- what about regions that
aren't in that enum? It seems like this still might be a problem for these
regions. I would imagine this is a problem that users have ran into, and
isn't being addressed in either PR.

**Comment by zenfenan** [comment]
@jmlogan It is an enum on the AWS SDK side so whatever new regions that are added to AWS cloud services, the team managing the SDK project update the enum on their side.

**Comment by jmlogan** [comment]
There are non public regions not in that list.

**Comment by pvillard31** [comment]
Closing this PR as the original intent has been resolved with #3190. @jmlogan, if you feel like there is something to be done for additional regions that are not exposed by the AWS SDK, feel free to open a JIRA / submit a PR for that.
